local-volume/provisioner/deployment/kubernetes/helm/provisioner/templates/provisioner.yaml

```diff
@@ -0,0 +1,70 @@
+{{- $configMapName := .Values.configmap.configMapName }}
+{{- $provisionerNamespace := .Values.common.namespace }}
+{{- $provisionerName := .Values.daemonset.name }}
+{{- $imageFull := .Values.daemonset.image }}
+{{- $imagePullPolicy := .Values.daemonset.imagePullPolicy }}
+{{- $containerConfigDirectory := .Values.daemonset.containerConfigDirectory }}
+{{- $defaultMountDir := .Values.common.defaultMountDir }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ $configMapName }} 
+  namespace: {{ $provisionerNamespace }} 
+data:
+  storageClassMap: |
+{{- range $tmp, $val := .Values.configmap.storageClass }}
+   {{- range $key, $class := $val }}
+      {{ $key }}:     
+         {{- range $val1, $val2 := $class }}
+           {{ $val1 }}: {{ $val2 }}
+      {{- end}}
+   {{- end}}
+{{- end}}
+
+---
+
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: {{ $provisionerName }}
+  namespace: {{ $provisionerNamespace }}
+  labels:
+    app: local-volume-provisioner
+spec:
+  selector:
+    matchLabels:
+      app: local-volume-provisioner 
+  template:
+    metadata:
+      labels:
+        app: local-volume-provisioner
+    spec:
+      nodeSelector:
+      containers:
+        - image: "{{ $imageFull }}"
+          imagePullPolicy: {{ $imagePullPolicy | quote }}
+          name: provisioner 
+          securityContext:
+            privileged: true
+          env:
+          volumeMounts:
+            - mountPath: {{ $containerConfigDirectory }}
+              name: provisioner-config
+              readOnly: true
+{{- range $tmp, $val := .Values.configmap.storageClass }}
+   {{- range $key, $class := $val }}
+            - mountPath: {{ if $class.mountDir }} {{ $class.mountDir }} {{ else }} {{ $defaultMountDir }} {{ end }}
+              name: {{ $key }}
+   {{- end}}
+{{- end}}
+      volumes:
+        - name: provisioner-config
+          configMap:
+            name: {{ $configMapName }}
+{{- range $tmp, $val := .Values.configmap.storageClass }}
```

Comment:
Add a comment here saying these are all the hostpath volumes for volume discovery per storageclass

Reply:
done